library/psa_crypto.c

```diff
@@ -4768,26 +5059,23 @@ static psa_status_t psa_key_derivation_input_raw(
     return( status );
 }
 
-psa_status_t psa_key_derivation_input_bytes( psa_key_derivation_operation_t *operation,
-                                             psa_key_derivation_step_t step,
-                                             const uint8_t *data,
-                                             size_t data_length )
+psa_status_t psa_key_derivation_input_bytes(
+    psa_key_derivation_operation_t *operation,
+    psa_key_derivation_step_t step,
+    const uint8_t *data,
+    size_t data_length )
 {
-    switch( step )
-    {
-        case PSA_KEY_DERIVATION_INPUT_LABEL:
-        case PSA_KEY_DERIVATION_INPUT_SALT:
-        case PSA_KEY_DERIVATION_INPUT_INFO:
-            return( psa_key_derivation_input_raw( operation, step,
-                                                  data, data_length ) );
-        default:
-            return( PSA_ERROR_INVALID_ARGUMENT );
-    }
+    if( step == PSA_KEY_DERIVATION_INPUT_SECRET )
```

Comment:
Blacklisting instead of whitelisting inputs that can be direct buffers probably saves a few bytes of code, but it's more fragile. I'd prefer to keep whitelisting.

Reply:
Saving a couple of bytes wasn't the only reason for removing it.

My thinking was that the specific functions called inside `psa_key_derivation_input_internal()` need to check this again anyway. This introduces redundancy that makes maintenance more error prone.

Also checking a whitelist here gives limited value. For example if the user calls this for a TLS 1.2 PRF operation with `PSA_KEY_DERIVATION_INPUT_INFO` we still depend on the specialised input function to catch it. This means that even though we checked it here, we still can't make any useful assumption about this parameter later down in the call stack.

Reply:
Ok, good point.